File: backend/src/main/java/com/urmyfood/backend/application/dto/PostResponse.java
package com.urmyfood.backend.application.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PostResponse {

    @JsonProperty("post_id")
    private UUID postId;

    @JsonProperty("dish_name")
    private String dishName;

    private BigDecimal price;

    @JsonProperty("original_price")
    private BigDecimal originalPrice;

    @JsonProperty("max_quantity")
    private int maxQuantity;

    @JsonProperty("remaining_quantity")
    private int remainingQuantity;

    @JsonProperty("end_time")
    private OffsetDateTime endTime;

    @JsonProperty("is_flash_sale")
    private boolean isFlashSale;

    private String status;

    private String content;

    @JsonProperty("image_url")
    private String imageUrl;

    @JsonProperty("shop_name")
    private String shopName;

    @JsonProperty("shop_avatar_url")
    private String shopAvatarUrl;

    @JsonProperty("shop_address")
    private String shopAddress;

    @JsonProperty("created_at")
    private OffsetDateTime createdAt;

    @JsonProperty("like_count")
    private long likeCount;

    @JsonProperty("is_liked")
    private boolean isLiked;

    @JsonProperty("comment_count")
    private long commentCount;

    @JsonProperty("category")
    private String category;
}

File: frontend/app-user/src/main/java/com/urmyfood/user/data/model/PostResponse.kt
package com.urmyfood.user.data.model

import com.google.gson.annotations.SerializedName

data class PostResponse(
    @SerializedName("post_id") val postId: String,
    @SerializedName("dish_name") val dishName: String,
    @SerializedName("price") val price: Double,
    @SerializedName("original_price") val originalPrice: Double,
    @SerializedName("max_quantity") val maxQuantity: Int,
    @SerializedName("remaining_quantity") val remainingQuantity: Int,
    @SerializedName("end_time") val endTime: String?,
    @SerializedName("is_flash_sale") val isFlashSale: Boolean,
    @SerializedName("status") val status: String,
    @SerializedName("content") val content: String?,
    @SerializedName("image_url") val imageUrl: String?,
    @SerializedName("shop_account_id") val shopAccountId: Long = 0L,
    @SerializedName("shop_name") val shopName: String,
    @SerializedName("shop_avatar_url") val shopAvatarUrl: String?,
    @SerializedName("shop_address") val shopAddress: String?,
    @SerializedName("created_at") val createdAt: String?,
    @SerializedName("like_count") val likeCount: Int = 0,
    @SerializedName("is_liked") val isLiked: Boolean = false,
    @SerializedName("is_following_shop") val isFollowingShop: Boolean = false,
    @SerializedName("is_saved") val isSaved: Boolean = false,
    @SerializedName("comment_count") val commentCount: Int = 0
)
